### DISCUSSION TITLE
Can't make interface general resolvers

### DISCUSSION THREAD

**Discussion by chav-aniket** [99designs/gqlgen#1896] [Q&A] [answered]
### What happened?
I have an interface for a stocklist. Portfolio and Watchlist both implement Stocklist. The Stocklist type has its own defined fields including name, id and a list of stocks. I want to make a generalised resolver for both portfolio and watchlist to be able to add a stock to the stocklist. 

### What did you expect?

So the variable I'm working with is of type stocklist, as that's what the find (from mongodb) function returns. Even though the fields are defined for stocklist however, the only thing I can apparently access is a `isStocklist()` method. I expected to be able to access the list of stocks to be able to append to them and remove from them.

### Minimal graphql.schema and models to reproduce

I do apologise for the poor naming, that the interface is called stocklist but the stocklists have a field called stocklist 😅, I'll try to name it something a bit better in the future

```gql
interface Stocklist {
  id: String!
  name: String!
  stockList: [String!]
}

type Portfolio implements Stocklist {
  id: String!
  name: String!
  stockList: [String!]
  isPublic: Boolean!
  total: Float!
  cashHolding: Float
  dayGain: Float
  totalGain: Float
  followedBy: [User!]
}

type Watchlist implements Stocklist {
  id: String!
  name: String!
  stockList: [String!]
}
```

### versions
 - `gqlgen version`? 
    - 0.14.0
 - `go version`
    - 1.17
 - dep or go modules:
    - mongo-driver
    - chi
    - gqlparser/v2

**Comment by duckbrain** [comment, marked as answer]
The Go interfaces generated by gqlgen will only contain the one method to identify them. You basically have two options.

1. Type assert to a concrete type. The `Stocklist` passed to your resolver is either a `Watchlist` or `Portfolio`. You can do something like the following to get the concrete value to work with.
```go
swtich v := obj.(type) {
case WatchList:
	// v is a WatchList here
case Portfolio:
	// v is a Portfolio here
}
```

2. Define the `Stocklist` interface yourself. gqlgen can pick up existing model types. You can define a different set of methods and the resolver will be able to access them. You'll just need to make sure you satisfy that interface for `Watchlist` and `Portfolio`. (Don't define things in the generated file.)

**Reply by chav-aniket** [reply]
Sorry for the late response but thank you for answering! This makes complete sense to me
